### ISSUE TITLE
error with phylosift search

### ISSUE THREAD

**Issue by ucassee** [gjospin/PhyloSift#501]
When I run `phylosift search -help`  it went error with following:

> NCBI taxonomy data not found and unable to connect to update server, please check your phylosift configuration and internet connection!
at /home/zhouyl/software/phylosift_v1.0.1/bin/../lib/Phylosift/Phylosift.pm line 154.

I downloaded markers.tgz and ncbi.tgz. Where should I set them?

Thanks in advance

**Comment by gjospin** [comment]
you can specify a custom directory in the phylosiftrc file, make sure you
remove the # at the beginning of the line that you are using.

The default place that PS will look for things is in:
<$HOME>/share/phylosift

**Comment by ucassee** [comment]
@gjospin Thanks for your reply.
The phylosift is at `/home/zhouyl/software/phylosift_v1.0.1/bin/` . I move ncbi.tgz to `/home/zhouyl/software/phylosift_v1.0.1/`  . But it doesn't work. It still with error 

> NCBI taxonomy data not found and unable to connect to update server, please check your phylosift configuration and internet connection!

**Comment by gjospin** [comment]
you need your path to look like

/home/zhouyl/share/phylosift/ncbi
/home/zhouyl/share/phylosift/markers

**Comment by ucassee** [comment]
@gjospin  Thanks for your patience~
If I want to change the location of these two database, can I modify the code ` /home/zhouyl/software/phylosift_v1.0.1/bin/../lib/Phylosift/Phylosift.pm` ? But in the line 154 I only see `Phylosift::Utilities::data_checks( self => $self )`

**Comment by gjospin** [comment]
Change /home/zhouyl/software/phylosift_v1.0.1/phylosiftrc

find the line

# $marker_dir="";

change to

$marker_dir="/home/zhouyl/software/phylosift_v1.0.1/markers";


# $ncbi_dir

change to

$ncbi_dir="/home/zhouyl/software/phylosift_v1.0.1/ncbi"

**Comment by ucassee** [comment]
When I use phylosift to search conservative protein. I find there are too much same proteins in one of my genomes.

> -rw-r--r-- 1 zhouyl microbial 1515 Jul 12 12:31 DNGNGWU00013.lastal.candidate.aa.1.10
-rw-r--r-- 1 zhouyl microbial 1440 Jul 12 12:31 DNGNGWU00013.lastal.candidate.aa.1.11
-rw-r--r-- 1 zhouyl microbial 1151 Jul 12 12:31 DNGNGWU00013.lastal.candidate.aa.1.14
-rw-r--r-- 1 zhouyl microbial  377 Jul 12 12:31 DNGNGWU00013.lastal.candidate.aa.1.15
-rw-r--r-- 1 zhouyl microbial  761 Jul 12 12:31 DNGNGWU00013.lastal.candidate.aa.1.16
-rw-r--r-- 1 zhouyl microbial  750 Jul 12 12:31 DNGNGWU00013.lastal.candidate.aa.1.19
-rw-r--r-- 1 zhouyl microbial  771 Jul 12 12:31 DNGNGWU00013.lastal.candidate.aa.1.2
-rw-r--r-- 1 zhouyl microbial  389 Jul 12 12:31 DNGNGWU00013.lastal.candidate.aa.1.20
-rw-r--r-- 1 zhouyl microbial  379 Jul 12 12:31 DNGNGWU00013.lastal.candidate.aa.1.3
-rw-r--r-- 1 zhouyl microbial 1138 Jul 12 12:31 DNGNGWU00013.lastal.candidate.aa.1.4
-rw-r--r-- 1 zhouyl microbial 1112 Jul 12 12:31 DNGNGWU00013.lastal.candidate.aa.1.5
-rw-r--r-- 1 zhouyl microbial  373 Jul 12 12:31 DNGNGWU00013.lastal.candidate.aa.1.6
-rw-r--r-- 1 zhouyl microbial  773 Jul 12 12:31 DNGNGWU00013.lastal.candidate.aa.1.7
-rw-r--r-- 1 zhouyl microbial  413 Jul 12 12:31 DNGNGWU00013.lastal.candidate.aa.1.8
-rw-r--r-- 1 zhouyl microbial 1159 Jul 12 12:31 DNGNGWU00013.lastal.candidate.aa.1.9

But when I use the protein sequences in these files to blast on ncbi, there is no significant similarity found. I fell confused. Could you help me ?

**Comment by ucassee** [comment]
When I see the `marker_summary.txt`. 
>DNGNGWU00010    1
DNGNGWU00011    1
DNGNGWU00012    1
DNGNGWU00013    35
DNGNGWU00014    1
DNGNGWU00015    1
DNGNGWU00016    1

I think it is unlikely to have  35  `DNGNGWU00013` proteins in the genome. And combined with no hit in ncbi, I guess whether it is a mistake in the search progress.

**Comment by ucassee** [comment]
I have another another question. I use the the sequences in the ` *candidate.ffn*` file to blast with the original genome sequences. I find when compared with original genome sequences, there are few gaps  in the sequences in ` *candidate.ffn*` file. If the ffn was  extracted from the original genome, why are these gaps exist?

**Comment by gjospin** [comment]
The search step is really permissive in the matches. It is really meant to
reduce the complexity of the alignment step so it doesn't take too long. I
would perform the align step and see what sticks after that.
Also keep in mind that PS was developed with short reads in mind. You may
want to adjust thresholds in the phylosiftrc file in the same way you
modified the paths yesterday. The thresholds might need to be adjusted
depending on the length of the marker you are looking at.

How big are the gaps? Could it be some frameshift happening? The hits are
done in protein space.

I hope this helps.

**Comment by ucassee** [comment]
The gaps are not too big, just  3,6, or 9 bp in my sequences. 
Most of the predicted `DNGNGWU` protein can find homologous protein when blast in ncbi . But `DNGNGWU00013` I mentioned before could not. So do you suggest if I improve the threshold I can get more positive result?
I can't find the threshold setting line in `Phylosift.pm`. Could you help  me ?

**Comment by gjospin** [comment]
No, I was suggesting to increase your threshold stringency to remove the
incorrect matches that should have lower hits. So if a default score is 150
in the phylosiftrc file, you would want to increase that to filter out
false positives. You could enforce a minimum number of bases getting
aligned also.

We have seen clades not have certain markers. It's possible marker 13
doesn't work well for your bug of interest. If the matches aren't good
enough, then this marker13 space in the concatenated alignment would be all
gaps. I would ignore it if you aren't happy with what comes out of it.

There is a way to give PS a list of markers (--custom flag), 1 per line to
only look at matches for markers in the list. You could give it the list of
37markers minus DNGNGWU00013

**Comment by ucassee** [comment]
Hi@jospin, Thanks for your hfelp. In the phylosiftrc file, I find too much parameters. I am not sure which one should I modify to increase the threshold. Is it the following ?

> # MarkerAlign default parameters
#$min_aligned_residues=50;
#$rna_split_size = 500; #sequences longer than this value will undergo the long sequence pipeline
#$gap_character = "-";

**Comment by gjospin** [comment]
I would target the $min_aligned_residues and extend that closer to the
gene(s) you are interested in.
You can find the length markers in the database's HMM files. (grep 'LEN'
DNGNG*/*.hmm for example, our system is down right now, so I can't check
the exact syntax).
Also keep in mind this is in AA space, so 50 represents 150 nucleotides.
